### PR TITLE
LibWeb: Hide inner text of input element when showing placeholder

### PR DESCRIPTION
**Pull request by bplaat**
The placeholder in the input element was not completly visible, this pr fixes that by hiding the inner text element when the placeholder is visible.

## Before
Screenshot 2024-03-11 at 19 05 35

## After
Screenshot 2024-03-11 at 19 05 13

### CODE REVIEW

**Comment by awesomekling** [comment]
Can we add a test?

File: Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp [hunk 1]
```diff
@@ -648,15 +649,15 @@ String HTMLInputElement::placeholder() const
 }
 
 // https://html.spec.whatwg.org/multipage/input.html#attr-input-placeholder
-Optional<ByteString> HTMLInputElement::placeholder_value() const
+Optional<String> HTMLInputElement::placeholder_value() const
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp:652]: Converting to a String should also be in its own commit

**Comment by bplaat** [comment]
I've added a Layout test
